Remove obsolete Makefile.win and Makefile.msys2

klaasvanaarsen opened the pull request:
The files Makefile.msys2 and Makefile.msys2.sdl2 are superseded by cmake.
The file Makefile.win was not referred to in `docs/hacking/Compiling.rst` nor anywhere else.

backwardsEric commented:
I agree that Makefile.msys2 is now obsolete:  what it did can be done via CMake or autotools and installing CMake is not a significant barrier in MSYS2.

Makefile.msys2.sdl2 statically links in the SDL2 library and all of its dependencies that are not standard Windows libraries.  Can that readily be done with CMake (perhaps by setting LDFLAGS)?  If so, then it might be good to document that in compiling.rst and have a workflow for MSYS2 and CMake that exercises statically linking SDL2.  If not, then Makefile.msys2.sdl2 is not obsolete.  There's been some interest in Makefile.msys2.sdl2 so dropping it without a mostly equivalent replacement could generate some blowback.

Besides Nick's comment, there are bits in Makefile.win that claim to be adapted to Cygwin and PDCurses.  Before dropping Makefile.win, I'd want to verify that a build using CMake or autotools was relatively straightforward on Cygwin and that PDCurses could be used with CMake or autotools.

NickMcConnell commented:
I seem to recall I cygwin was kind of deprecated at some point (probably 10-15 years ago; certainly I moved to using msys.  It may have been a licencing issue for builds with cygwin.  Sorry I don't have better info.

klaasvanaarsen commented:
## Cygwin
- I have added a github action to validate cygwin. It builds with both cmake ("Stock") and autotools, and validates if we can run angband.
- I haven't added `Makefile.win` to the github action, since it's not clear to me how it is supposed to be used on cygwin.
- Autotools currently fails because running angband fails immediately with an exit code `127`.

## Makefile.msys2.sdl2
- I have restored `Makefile.msys2.sdl2` for now and added validation for it in the msys2 github action.
- I've set both the cmake version and the Makefile version to verbose so we can compare the flags.
- Interestingly, quite a few flags are missing from the cmake build, which needs some further analysis.

NickMcConnell commented:
I'm holding fire on this one as it looks like there's still stuff to do?

klaasvanaarsen commented:
> I'm holding fire on this one as it looks like there's still stuff to do?

Yes.
I think it is mergeable as it is, but there are a couple of things I believe I should still do within the scope of this PR:
1. Get static linking to work with cmake for the SDL2 target.
2. If successful, delete Makefile.msys2.sdl2 again.

klaasvanaarsen commented:
Cmake actually already knows what those libraries are:
```
-- Found PkgConfig: C:/msys64/mingw64/bin/pkg-config.exe (found version "2.5.1")
-- SDL2_LIBRARIES              = mingw32;SDL2main;SDL2
-- SDL2_STATIC_LIBRARIES       = mingw32;SDL2main;SDL2;m;kernel32;user32;gdi32;winmm;imm32;ole32;oleaut32;version;uuid;advapi32;setupapi;shell32;dinput8
-- SDL2_TTF_LIBRARIES          = SDL2_ttf;mingw32;SDL2main;SDL2
-- SDL2_TTF_STATIC_LIBRARIES   = SDL2_ttf;mingw32;SDL2main;SDL2;m;kernel32;user32;gdi32;winmm;imm32;ole32;oleaut32;version;uuid;advapi32;setupapi;shell32;dinput8;freetype;bz2;png16;z;harfbuzz;m;usp10;gdi32;rpcrt4;user32;dwrite;glib-2.0;intl;ws2_32;ole32;winmm;shlwapi;uuid;atomic;m;pcre2-8;graphite2;brotlidec;brotlicommon
-- SDL2_IMAGE_LIBRARIES        = SDL2_image;mingw32;SDL2main;SDL2
-- SDL2_IMAGE_STATIC_LIBRARIES = SDL2_image;mingw32;SDL2main;SDL2;m;kernel32;user32;gdi32;winmm;imm32;ole32;oleaut32;version;uuid;advapi32;setupapi;shell32;dinput8;png16;z;jxl;m;stdc++;hwy;brotlienc;brotlidec;brotlicommon;jxl_cms;m;stdc++;lcms2;lcms2_fast_float;m;tiff;jbig;zstd;lzma;Lerc;jpeg;def
```
I'm currently trying to convince cmake that is should use them.
